burn-import/src/pytorch/recorder.rs

```diff
@@ -0,0 +1,136 @@
+}
+
+impl<PS: PrecisionSettings, B: Backend> Recorder<B> for PyTorchFileRecorder<PS> {
+    type Settings = PS;
+    type RecordArgs = PathBuf;
+    type RecordOutput = ();
+    type LoadArgs = LoadArgs;
+
+    fn save_item<I: Serialize>(
+        &self,
+        _item: I,
+        _file: Self::RecordArgs,
+    ) -> Result<(), RecorderError> {
+        unimplemented!("save_item not implemented for PyTorchFileRecorder")
+    }
+
+    fn load_item<I: DeserializeOwned>(&self, _file: Self::LoadArgs) -> Result<I, RecorderError> {
+        unimplemented!("load_item not implemented for PyTorchFileRecorder")
+    }
+
+    fn load<R: Record<B>>(
+        &self,
+        args: Self::LoadArgs,
+        device: &B::Device,
+    ) -> Result<R, RecorderError> {
+        let item = from_file::<PS, R::Item<Self::Settings>, B>(&args.file, args.key_remap)?;
+        Ok(R::from_item(item, device))
+    }
+}
+
+/// Arguments for loading a PyTorch file.
+///
+/// # Fields
+///
+/// * `file` - The path to the file to load.
+/// * `key_remap` - A vector of tuples containing a regular expression and a replacement string.
+///                See [regex::Regex::replace](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace)
+///                for more information.
+///
+/// # Notes
+///
+/// Use [Netron](https://github.com/lutzroeder/netron) to inspect the keys of the PyTorch file (.pt extension).
+///
+///
+/// # Examples
+///
+/// ```text
+/// use burn_import::pytorch::{LoadArgs, PyTorchFileRecorder};
+/// use burn::record::FullPrecisionSettings;
+/// use burn::record::Recorder;
+///
+/// let args = LoadArgs::new("tests/key_remap/key_remap.pt".into())
+///    .with_key_remap("conv\\.(.*)", "$1"); // // Remove "conv" prefix, e.g. "conv.conv1" -> "conv1"
+///
+/// let record = PyTorchFileRecorder::<FullPrecisionSettings>::default()
+///   .load(args)
+///   .expect("Should decode state successfully");
+/// ```
+#[derive(Debug, Clone)]
+pub struct LoadArgs {
+    /// The path to the file to load.
+    pub file: PathBuf,
+
+    /// A list of key remappings.
+    pub key_remap: Vec<(Regex, String)>,
+}
+
+impl LoadArgs {
+    /// Create a new `LoadArgs` instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `file` - The path to the file to load.
+    pub fn new(file: PathBuf) -> Self {
+        Self {
+            file,
+            key_remap: Vec::new(),
+        }
+    }
+
+    /// Set key remapping.
+    ///
+    /// # Arguments
+    ///
+    /// * `pattern` - The Regex pattern to be replaced.
+    /// * `replacement` - The pattern to replace with.
+    ///
+    /// See [Regex](https://docs.rs/regex/1.5.4/regex/#syntax) for the pattern syntax and
+    /// [Replacement](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace) for the
+    /// replacement syntax.
+    pub fn with_key_remap(mut self, pattern: &str, replacement: &str) -> Self {
+        let regex = Regex::new(&format!("^{}$", pattern)).unwrap();
```

Comment:
I don't think it's a good idea to insert `^` and `$` in this place, it will mislead the user.
And there are many use cases where `^` and `$` need to be removed. For example I tried to rename all keys like `conv.0` to `conv0` because I used some workaround to implement PyTorch's `Sequential` in Burn. I want to do this for the whole model, it would be convenient if I could use `.with_key_remap(r#"([a-z]+)\.(\d+)"#, "$1$2")`.

Reply:
Sounds good. If you could submit a quick PR or issue, we will merge/fix it. 

Thanks for letting us know.

Reply:
@Nikaidou-Shinku I submitted a PR fix: https://github.com/tracel-ai/burn/pull/1196